manager/src/main/java/org/openremote/manager/event/ClientEventService.java

```diff
@@ -162,68 +173,92 @@ public void configure() throws Exception {
                             .to(GatewayService.GATEWAY_EVENT_TOPIC)
                         .endChoice()
                         .stop()
+                    .when(bodyAs(String.class).startsWith(EventSubscription.SUBSCRIBE_MESSAGE_PREFIX))
+                        .process(exchange -> exchange.getIn().setHeader(HEADER_CONNECTION_TYPE, HEADER_CONNECTION_TYPE_WEBSOCKET))
+                        .to(ClientEventService.CLIENT_EVENT_QUEUE)
+                        .stop()
+                    .when(bodyAs(String.class).startsWith(CancelEventSubscription.MESSAGE_PREFIX))
+                        .to(ClientEventService.CLIENT_EVENT_QUEUE)
+                        .stop()
+                    .when(bodyAs(String.class).startsWith(RenewEventSubscriptions.MESSAGE_PREFIX))
+                        .to(ClientEventService.CLIENT_EVENT_QUEUE)
+                        .stop()
+                    .when(bodyAs(String.class).startsWith(SharedEvent.MESSAGE_PREFIX))
+                        .convertBodyTo(SharedEvent.class)
+                        .process(exchange -> {
+                            SharedEvent event = exchange.getIn().getBody(SharedEvent.class);
+                            // If there is no timestamp in event, set to system time
+                            if (event.getTimestamp() <= 0) {
+                                event.setTimestamp(timerService.getCurrentTimeMillis());
+                            }
+                        })
+                        .choice()
+                            .when(exchange -> isGatewayClientId(getClientId(exchange)))
+                                .to(GatewayService.GATEWAY_EVENT_TOPIC)
+                            .otherwise()
+                                .to(ClientEventService.CLIENT_EVENT_TOPIC)
+                        .endChoice()
+                    .otherwise()
+                        .process(exchange -> LOG.fine("Unsupported message body: " + exchange.getIn().getBody()))
+                    .end();
+
+                from(ClientEventService.CLIENT_EVENT_QUEUE)
```

Comment:
This route is for outbound messages to `clients` shouldn't be `SUBSCRIBE` messages going in this direction....the only change to this route should be that after split() the `HEADER_CONNECTION_TYPE` should be checked and messages either routed to MQTT or Websocket...`splitForSubscribers` should be updated to add this header to each split message

Reply:
Actually ok to use this route for inbound and outbound message processing can always adjust camel routes later if we need to but my previous comment still applies about `splitForSubscribers`

Reply:
Move `SUBSCRIBE`, `CANCEL`, `RENEW` logic into `CLIENT_EVENT_QUEUE` processing of these messages shouldn't matter where they come from

Reply:
`routeId` should be updated also as not just for `websocket` messages any more